scene/gui/rich_text_label.cpp

```diff
@@ -1811,23 +1811,51 @@ void RichTextLabel::_notification(int p_what) {
 				ofs.y += main->lines[from_line].text_buf->get_size().y + main->lines[from_line].text_buf->get_line_count() * theme_cache.line_separation;
 				from_line++;
 			}
+
+			// Draw drag icon.
+			if (in_mb_scroll) {
+				theme_cache.scroll->draw_rect(ci, Rect2(mb_scroll_pos - theme_cache.scroll->get_size() / 2, theme_cache.scroll->get_size()));
+			}
 		} break;
 
 		case NOTIFICATION_INTERNAL_PROCESS: {
 			if (is_visible_in_tree()) {
+				double dt = get_process_delta_time();
+				if (in_mb_scroll) {
+					float delta = get_local_mouse_position().y - mb_scroll_pos.y;
+					vscroll->scroll(mb_scroll_rate * delta * dt * theme_cache.normal_font->get_height(theme_cache.normal_font_size));
+				}
 				if (!is_ready()) {
 					return;
 				}
-				double dt = get_process_delta_time();
-				_update_fx(main, dt);
-				queue_redraw();
+				if (has_active_eq) {
+					_update_fx(main, dt);
+					queue_redraw();
+				}
 			}
 		} break;
 
 		case NOTIFICATION_FOCUS_EXIT: {
 			if (deselect_on_focus_loss_enabled) {
 				deselect();
 			}
+			if (in_mb_scroll) {
+				in_mb_scroll = false;
+				if (!has_active_eq && !in_mb_scroll) {
```

Comment:
```suggestion
				if (!has_active_eq) {
```
Unnecessary because of the line above (or did you mean to put this above before the value was reset?